File: TripsServer/Trips/Entities/Trip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Trips.Entities
{
    public class Trip
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime? Date { get; set; }
        public Picture TitlePicture { get; set; }
        public Gallery Gallery { get; set; }

        public ICollection<UsersToTrips> Participants { get; set; }
        public ICollection<VehiclesToTrips> Vehicles { get; set; }
        public ICollection<Visit> Visits { get; set; }
    }
}

File: trips-client/src/app/models/trip.ts
import { Picture } from './picture';
import { Gallery } from './gallery';
import { UserHeader } from './user-header';
import { Visit } from './visit';
import { VehicleHeader } from './vehicle-header';

export class Trip {
    id: number;
    title: string;
    description: string;
    date: Date;
    titlePicture: Picture;
    gallery: Gallery;
    addedBy: UserHeader;
    addedDate: Date;
    changedBy: UserHeader;
    changedDate: Date;
    participants: UserHeader[];
    vehicles: VehicleHeader[];
    visits: Visit[];
}